### PR TITLE
alignment with AVM specification & grept-apply requirements

### PR DESCRIPTION
**Pull request by kewalaka**
## Description

This re-aligns the template with changes to the docs / AVM specification & fixes reported by grept-apply.

There are some existing PRs which repeat the same functionality, adding tags & new files to .gitignore based on grept-apply outputs, I believe this is more comprehensive (cc @jaredfholgate for a check)

It introduces the fix for private endpoints conflicting with Azure Policy based on the spec, with some corrections raised with @matt-FFFFFF here; https://github.com/Azure/terraform-azurerm-avm-res-keyvault-vault/issues/32#issuecomment-2050367744

I've used this as the starting point for the SQL Module that I'm preparing to post, as you can't properly lint this code due to things like TODOs.

-->

## Type of Change

- [X] Non-module change (e.g. CI/CD, documentation, etc.)
- [ ] Azure Verified Module updates:
  - [ ] Bugfix containing backwards compatible bug fixes, and I have NOT bumped the MAJOR or MINOR version in `locals.version.tf.json`:
    - [ ] Someone has opened a bug report issue, and I have included "Closes #{bug_report_issue_number}" in the PR description.
    - [ ] The bug was found by the module author, and no one has opened an issue to report it yet.
  - [ ] Feature update backwards compatible feature updates, and I have bumped the MINOR version in `locals.version.tf.json`.
  - [ ] Breaking changes and I have bumped the MAJOR version in `locals.version.tf.json`.
  - [ ] Update to documentation

# Checklist

- [X] I'm sure there are no other open Pull Requests for the same update/change (there are some minors - noted above).
- [ ] My corresponding pipelines / checks run clean and green without any errors or warnings
- [ ] I did run all  [pre-commit](https://azure.github.io/Azure-Verified-Modules/contributing/terraform/terraform-contribution-flow/#5-run-pre-commit-checks) checks

### CODE REVIEW

**Comment by kewalaka** [comment]
this also includes the fix for the location / "evil data source" 😅

File: .gitignore [hunk 1]
```diff
@@ -39,8 +39,9 @@ avmmakefile
 README-generated.md
 avm.tflint.hcl
 avm.tflint_example.hcl
+avm.tflint.merged.hcl
+avm.tflint_example.merged.hcl
 *tfplan*
 *.md.tmp
-
```
Comment (segraef) [inline .gitignore:44]: This disturbs me, leave my space please :D

Comment (kewalaka) [reply .gitignore:44]: that's what grept-apply said to do, i have no animosity towards blank lines 😂

**Review by jaredfholgate — APPROVED** [review]
LGTM

**Review by matt-FFFFFF — APPROVED** [review]

**Pull request merged**